src/figuresClasses.ts

```diff
@@ -1,19 +1,84 @@
+type Shapes = 'triangle' | 'circle' | 'rectangle';
+type Colors = 'red' | 'green' | 'blue';
+
 export interface Figure {
+  shape: Shapes;
+  color: Colors;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shapes = 'triangle';
+
+  constructor(
+    public color: Colors,
+    public a:number,
+    public b:number,
+    public c:number,
+  ) {
+    if (!this.isValid()) {
+      throw new Error('Not a valid triangle');
+    }
```

Comment:
you can hanlde error throwing right inside the isValid method.
Just have it in the constructor like `this.isValid()` - and everything will be done there